crates/router/src/connector/utils.rs

```diff
@@ -453,17 +457,46 @@ impl CardData for api::Card {
         ))
     }
     fn get_expiry_date_as_yyyymm(&self, delimiter: &str) -> Secret<String> {
-        let mut x = self.card_exp_year.peek().clone();
-        if x.len() == 2 {
-            x = format!("20{}", x);
-        }
+        let year = self.get_expiry_year_4_digit();
         Secret::new(format!(
             "{}{}{}",
-            x,
+            year.peek(),
             delimiter,
             self.card_exp_month.peek().clone()
         ))
     }
+
+    fn get_first_name_and_last_name_from_cardholdername(
+        &self,
+    ) -> Result<(Secret<String>, Secret<String>), errors::ConnectorError> {
+        let card_holder_name = self.card_holder_name.peek();
+        let words: Vec<&str> = card_holder_name.split_whitespace().collect();
+        if !words.is_empty() {
+            let first_name = Secret::new(if words.len() > 1 {
+                words[..words.len() - 1].join(" ")
+            } else {
+                words.first().unwrap_or(&"").to_string()
+            });
+            let last_name = Secret::new(if words.len() > 1 {
+                words.last().unwrap_or(&"").to_string()
```

Comment:
```suggestion
                words.last().unwrap_or_default()
```

Reply:
we can't use `unwrap_or_default()` here as the trait bound `&&str: std::default::Default` is not satisfied